File: ActivityBuilderDemo/app/src/main/java/com/tkm/activitybuilderdemo/UserActivity.java
package com.tkm.activitybuilderdemo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

import com.tkm.activitybuilder.annotations.Builder;
import com.tkm.activitybuilder.annotations.Args;

@Builder
public class UserActivity extends AppCompatActivity {
    private static final String TAG = "TAG";

    @Args
    private String name;

    @Args
    private String owner;

    @Args
    private String url;

    @Args
    private long createAt;

    private boolean isNormal;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_user);

        Bundle extras = getIntent().getExtras();
        name = extras.getString("NAME");
        owner = extras.getString("OWNER");
        url = extras.getString("URL");
        createAt = extras.getLong("CREATEAT");

        Log.d(TAG, "name: " + name +
                ", owner: " + owner +
                ", url: " + url +
                ", createAt: " + createAt);
    }
}

File: APTDemo/app/src/main/java/com/tkm/aptdemo/UserActivity.kt
package com.tkm.aptdemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.tkm.aptdemo.annotations.Builder
import com.tkm.aptdemo.annotations.Optional
import com.tkm.aptdemo.annotations.Required

@Builder
class UserActivity : AppCompatActivity() {
    @Required
    lateinit var name: String

    @Required
    lateinit var owner: String

    @Optional(stringValue = "")
    lateinit var url: String

    @Optional
    var createAt: Long = 0L

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_user)
    }
}